src/compiler.ts

```diff
@@ -2994,6 +2991,8 @@ export class Compiler extends DiagnosticEmitter {
           );
           pendingElements.delete(dummy);
           flow.freeScopedDummyLocal(name);
+
+          initType = this.currentType;
```

Comment:
Due to compiling with `Constraincts.CONV_IMPLICIT`, I think `initType` is always the same as `type` here. Sure that the additional `initType` variable is needed?

Reply:
consider this testcase. Actually `initType` and `type` is not always same
```tyepscript
export function testInit(a: Ref): void {
  let c: Ref | null = requireNonNull(a);
  if (isNullable(c)) ERROR("should be non-nullable");
}
```